src/sls_core.js

```diff
@@ -437,3 +491,176 @@ function createCollectionSlide(deck, insightDeck, slideLayout) {
+ * Finally, the function fetches the image shapes and ranges for the slide from
+ * the spreadsheet. If there are image shapes and ranges, the function adds the
+ * images to the slide.
+ *
+ *
+ * @param {SlidesApp.Presentation} deck The presentation to add the slide to.
+ * @param {SlidesApp.InsightDeck} insightDeck The insight deck that contains the
+ *     data for the slide.
+ * @param {SlidesApp.SlideLayout} slideLayout The layout to use for the slide.
+ *
+ * @return {void}
+ */
+function createSingleSlide(deck, insightDeck, slideLayout) {
+  const spreadsheet = SpreadsheetApp.getActive().getSheetByName(
+      documentProperties.getProperty('DATA_SOURCE_SHEET'));
+
+  const slide = deck.appendSlide(slideLayout);
+  if (deck.getMasters().length > 1) {
+    deck.getMasters()[deck.getMasters().length - 1].remove();
+  }
+
+  // Fetch fields
+  const titleRange = documentProperties.getProperty('TITLE_RANGE');
+  if (titleRange && titleRange.length > 0) {
+    const title = spreadsheet.getRange(titleRange).getValue();
+    const slideTitlePlaceholder =
+        slide.getPlaceholder(SlidesApp.PlaceholderType.TITLE);
+    const slideTitle = slideTitlePlaceholder.asShape().getText();
+    slideTitle.setText(title);
+  }
+
+
+  const subtitleRange = documentProperties.getProperty('SUBTITLE_RANGE');
+  if (subtitleRange && subtitleRange.length > 0) {
+    const subtitle = spreadsheet.getRange(subtitleRange).getValue();
+    const slideSubtitlePlaceholder =
+        slide.getPlaceholder(SlidesApp.PlaceholderType.SUBTITLE);
+    const slideSubtitle = slideSubtitlePlaceholder.asShape().getText();
+    slideSubtitle.setText(subtitle);
+  }
+
+  const bodyRange = documentProperties.getProperty('BODY_RANGE');
+  if (bodyRange && bodyRange.length > 0) {
+    const body = spreadsheet.getRange(bodyRange).getValue();
+    const slideBodyPlaceholder =
+        slide.getPlaceholder(SlidesApp.PlaceholderType.BODY);
+    const slideBody = slideBodyPlaceholder.asShape().getText();
+    slideBody.setText(body);
+  }
+
+  const imageShapesArray = documentProperties.getProperty('IMAGE_SHAPES')
+      .split(',')
+      .map((item) => item.trim());
+  const imageRangesArray = documentProperties.getProperty('IMAGE_RANGES')
+      .split(',')
+      .map((item) => item.trim());
+
+  if (imageShapesArray && imageShapesArray.length > 0) {
+    for (let i = 0; i < imageShapesArray.length; i++) {
+      const shapeId = imageShapesArray[i];
+      const range = imageRangesArray[i];
+
+      if (shapeId && range) {
+        const imageShape = retrieveShape(slide, shapeId);
+        const imageValue = spreadsheet.getRange(range).getValue();
+        slide.insertImage(
+            imageValue, imageShape.getLeft(), imageShape.getTop(),
+            imageShape.getWidth(), imageShape.getHeight());
+      }
+    }
+  }
+}
+
+
+/**
+ * Creates a collection slide based on a slide layout and data from a specified
+ * row in a Google Sheet.
```

Comment:
The comment for the single slide above is pretty thorough. It might be nice to have a little more detail here to make it clear what the difference is. It's something about how the images are specified, but it's not clear what that difference is other than one uses a range and one uses columns.